### ISSUE TITLE
Parsing raw transactions is complicated in 2.x

### ISSUE THREAD

**Issue by alcuadrado** [ethereumjs/ethereumjs-monorepo#708]
I'm upgrading this library in [Buidler](https://github.com/nomiclabs/buidler/), and found that parsing raw (i.e. RLP encoded) transactions can be challenging.

If the transaction is signed using EIP155, you need to know its `chainId` to call the `Transaction` constructor. This requirement may not be realistic, as you have to parse the tx to get its `chainId`, and you parse it by calling that constructor.

Once you get the chainId, two things can happen:

1. Somewhat easy case: That chain is supported by `ethereumjs-common` and you can pass it in the `TransactionOptions` object. But how do you know if it is supported? I'm using this hack, that feels really fragile:

```ts
  const chains = require("ethereumjs-common/dist/chains");
  if (chains.chains.names[chainId] !== undefined) {
     transaction = new Transaction(tx, { chain: chainId });
  }
```

2. Harder case: the chain is not supported by `ethereumjs-common`. What should you do in that case? I'm doing this:

```ts
  } else {
    const { default: Common } = await import("ethereumjs-common");

    const common = Common.forCustomChain(
      "mainnet",
      {
        chainId,
        networkId: chainId
      },
      "petersburg"
    );

    transaction = new Transaction(tx, { common });
  }
```

I think we should improve this situation by:

1. Adding a static method to `Common.isStandardChain(chain: number|string): boolean`.

2. Add a static factory to `Transaction`. Something like `Transaction.parse(rawTx:Buffer, 
 ... ): Transaction`.  I'm not completely sure how this method should work, but I think it should be able to parse txs from standard chains without extra arguments. If a tx isn't from a standard chain, it should provide more meaningful errors.

What do you think?

**Comment by ryanio** [comment]
@alcuadrado do you think the static factories introduced in #812 resolve this issue?

It doesn't solve the problem of parsing a Transaction without knowing its chainId (initializing without a common), do you think we should add that functionality?

**Comment by alcuadrado** [comment]
TBH I can't think of a good solution. Maybe make common required? But it would be pretty annoying.

**Comment by ryanio** [comment]
Yeah right now common is required, as it does simplify things.

I was just pondering your use case of trying to parse a Transaction without knowing its chain (i.e. without a common).

**Comment by alcuadrado** [comment]
I meant here, where it's optional: https://github.com/ethereumjs/ethereumjs-vm/pull/812/files#diff-80443874f320e0ef1b23e3ba293dffbdR52

I just checked Buidler's codebase, and we don't have that case. We do know the chain id, by fetching it before parsing the tx. I guess that's a reasonable requirement.

**Comment by holgerd77** [comment]
This should have been significantly eased respectively solved with #1292, will close.
